docs/overview/architecture/README.md

```diff
@@ -14,36 +14,22 @@ execute and test governance proposals.
 ## Proposal Generic Contract
 
 At its core, the FPS features a **Proposal Contract**. This contract defines
-[external](external-functions.md) and [internal](internal-functions.md)
-functions, customizable by the protocol utilizing FPS. The `run` function serve as the entry point to execute a proposal.
+[proposal-functions](proposal-functions.md), customizable by the protocol utilizing FPS. The `run` function serve as the entry point to execute a proposal.
 
 ## Governance Specific Contracts
 
-FPS accommodates different Governance types (e.g., Timelock, Multisig) through specialized contracts. These Governance Specific Contracts inherit from the Proposal Contract, tailoring their functions to unique governance requirements.
+FPS accommodates different Governance types (e.g., Timelock, Multisig, Governor Bravo) through specialized contracts. These Governance Specific Contracts inherit from the Proposal Contract, tailoring their functions to unique governance requirements.
 
 ## Proposal Specific Contract
 
-Protocols using FPS must create their own Proposal Specific Contracts, conforming to FPS standards. These contracts override functions relevant to the particular proposal, such as `deploy()` and `_afterDeployMock()` for proposals involving new contract deployments. For more details, refer to [internal functions](internal-functions.md).
-
-## Test Suite
-
-An integral part of FPS, the Test Suite includes a `testProposals()`
-method. This contract interacts with the `Addresses` object, holding key addresses for the proposal process.
+Protocols using FPS must create their own Proposal Specific Contracts, conforming to FPS standards. These contracts override functions relevant to the particular proposal, such as `deploy()` and `afterDeployMock()` for proposals involving new contract deployments. For more details, refer to [proposal functions](proposal-functions.md).
 
 ## Post Proposal Check
 
-Protocols must implement a `Post Proposal Check` contract. This contract is tasked with deploying Proposal instances and invoking `TestSuite.testProposals()` for simulation purposes. Integration tests should inherit from `Post Proposal Check`.
-
-## Script Suite
-
-Serves as a multipurpose tool. Primarily used for proposal simulation, it can also execute proposals on-chain when used with the `--broadcast` flag. The Script Suite works alongside the `Addresses` object to manage proposal address references.
-
-{% content-ref url="external-functions.md" %}
-[external-functions.md](external-functions.md)
-{% endcontent-ref %}
+Protocols should implement a [Post Proposal Check](https://github.com/solidity-labs-io/fps-example-repo/tree/main/test/bravo/BravoPostProposalCheck.sol) contract. This contract is tasked with deploying Proposal instances and invoking `proposal.run()` for simulation purposes. Integration tests should inherit from `Post Proposal Check`.
```

Comment:
This is not part of FPS architecture. I believe we should move this to the guides section. Let's make things as simples as possible